runtime/str.go

```diff
@@ -806,6 +806,8 @@ func strCompare(v, w *Object, ltResult, eqResult, gtResult *Int) *Object {
 }
 
 func strInterpolate(f *Frame, format string, values *Tuple) (*Object, *BaseException) {
+	var val string
+	var err error
```

Comment:
I would declare these closer to where they're used. Especially err, which only seems to be used in one block.